README.md

```diff
@@ -85,7 +88,49 @@ Now let's get into some more details for each of the data types.
 
 ### Data
 
-The most straightforward of the options, this will just pass the data right through to the locals. Also if you provide a A+ compliant promise for a value, it will be resolved and passed in to the template.
+The most straightforward of the options, this will just pass the data right through to the locals. Also if you provide an A+ compliant promise for a value, it will be resolved and passed in to the template.
+
+#### Promises
+
+If you're new to promises, this [Google Developers](https://developers.google.com/web/fundamentals/getting-started/primers/promises) article should get you up to speed. The following example is similar to that article. Since we're using Webpack let's keep app.js clean by requiring our promise function from a module we'll create named `promiseModule`. You have the power of Node.js, Webpack, and Spike available to you in this module, but here's a simple example.
+
+##### promiseModule.js
+Here's our module, which exports one function that returns a promise that will either be resolved or rejected. This is the module that was required above.
+```js
+module.exports = function(obj) {
+  return new Promise((resolve, reject) => {
+    //As an example, let's create a new pizza menu showing their type and price.
+    let pizzas = new Array;
+    for(var prop in obj) {
+      if(${prop}.includes('Pizza')) {
+        pizzas.push({
+          type: ${prop}.replace("Pizza", ""),
+          price: "$".concat(${obj[prop]})
+        });
+      }
+    }
+    //Pizza Menu
+    if(pizzas.length > 0) {
+      resolve(pizzas);
+    } else {
+      reject(Error("There's no pizza :("));
+    }
+  });
+};
+```
```

Comment:
Sounds good! Generally, I think just saying "you can pass a function that returns a promise and it will work" is pretty sufficient for the docs for this library, without going into deep detail on what promises are or how to use them, since that's kind of out of scope for this specific library.

However, I think something like this would make a fantastic blog post that a lot of people could learn from, and which I would be happy to link out to from the docs, if you're into that!